RELEASE-CHECKLIST.md

```diff
@@ -0,0 +1,28 @@
+# Releasing `substrate-archive`
+
+Here's how to make a new release of `substrate-archive`.
+
+## Checklist
+
+1. Make a new branch, name it whatever you want, but `vx.y.z-prep` is a good template
+1. Update `Cargo.toml` in `substrate-archive` and `substrate-archive-backend` with the new version number.
+1. Update all references to `substrate` crates and `polkadot` to their respective latest releases.
+1. Update the `CHANGELOG` as specified [here](https://keepachangelog.com/en/1.0.0/).
+1. Run all tests: `TEST_DATABASE_URL="postgres://localhost:5432/archive cargo test --all`.
+1. Push the PR against the `release` branch.
+1. Once reviewed, merge it to `release`.
+1. Tag the `release` branch` with `git tag vx.y.z` and push the tags with `git push --tags`
+1. Build a binary for debian, compatible with the current glibc version (`v2.31`)
+    1. `docker run --rm -it debian:jessie` (requires at least 8Gb RAM)
+    1. install required dependencies `apt-get update && apt-get -y install git curl gcc clang`
+    1. install rust from rustup.rs: `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y`
+    1. reload the shell: `source $HOME/.cargo/env`
+    1. build archive: `git clone https://github.com/paritytech/substrate-archive.git --branch release --single-branch && cd substrate-archive/bin/polkadot-archive && SKIP_WASM_BUILD=1 cargo build --release`
+    1. keeping the container running, in a another terminal find the id of the docker container with `docker ps -a`, copy the binary to host with `docker cp $YOUR_CONTAINER_ID:/substrate-archive/bin/polkadot-archive/target/release/polkadot-archive .`
+1. Prepare a new draft release in the github UI
+    1. Upload the new binary
+    1. Tag the new release against the **release** branch
```

Comment:
Not against the release commit on the master branch to make sure all the release stuff is approved (and release notes etc make it in)?

Reply:
Right, we either have to sync the  branches or keep release notes & changelog only on the `release` branch. Since releasing now consists of changing all the Cargo tags to frozen substrate versions which is different than the versions on master.

So I think we have to split this into two PR's:
- one that adds the release checklist & changelog updates
- the other that updates the `release` branch with the right substrate versions

we would have to do this on every release too I think -- IE update changelog in one PR and then sync up `release` in the next. 

Another option is just keeping the Changelog & Release checklist only on the `release` branch. If we do that, though, we should definitely put that in the `README` somewhere because it could be confusing for anyone looking for the changelog.

Reply:
I was thinking I'd make a new PR to `master` with the release checklist and changelog updates. 

The next release would branch off of the `release` branch and merge `master` into it, then merge back to `release` I presume?

Reply:
Yes, that's how it would have to work I think